### ISSUE TITLE
Modx3 does not support upadting a resouce (ERROR 500)

### ISSUE THREAD

**Issue by AmaZili** [modxcms/revolution#15971]
### Summary
Quick summary what's this issue about.
Using Modx3 3.0.0 RC1 with diffrent php version (7.1, 8.0, 8.1) gives an error 500 when updating a resource
### Step to reproduce
How to reproduce the issue, including custom code if needed.
Create a resource, save it then publish and save
### Observed behavior
How it behaved after following steps above.
Never endind
### Expected behavior
How it should behave after following steps above.
Should save resource
### Related issue(s)/PR(s)
Let us know if this is related to any issue/pull request.

### Environment
MODX version, apache/nginx and version, mysql version, browser, etc. Any relevant information.

nginx version: nginx/1.20.1
Mariadb version 10.2.41

MODX Revolution 3.0.0-rc1 (traditional)
Nom de code de la version:
Revolution
phpinfo():
Voir
Heure du serveur:
05:09:41 PM
Heure locale:
05:09:41 PM
Décalage du serveur:
0
ExtJS Version:
3.4.1
Smarty Version:
4.0.3
PHPMailer Version:
6.5.3
MagpieRSS Version:
0.72
Type de base de données:
mysql
Version de la base de données:
10.2.41-MariaDB
Charset de la base de données:
utf8mb4
Nom de la base de données:
matthieu_
Serveur de la base de données:
localhost
Préfix de table:
modx_
PDO support:
enabled
PDO drivers:
mysql, odbc, pgsql, sqlite
PDO Driver for MySQL:
enabled
Client API version:
mysqlnd 7.4.27
Directive:
Local Value,Master Value
pdo_mysql.default_socket:
/var/lib/mysql/mysql.sock,/var/lib/mysql/mysql.sock
Zip:
enabled
Zip version:
1.15.6
Libzip headers version:
1.3.2
Libzip library version:
1.3.2
allow_url_fopen:
On
allow_url_include:
Off
arg_separator.input:
&
arg_separator.output:
&
auto_append_file:
no value
auto_globals_jit:
On
auto_prepend_file:
no value
browscap:
no value
default_charset:
UTF-8
default_mimetype:
text/html
disable_classes:
no value
disable_functions:
opcache_get_status
display_errors:
Off
display_startup_errors:
Off
doc_root:
no value
docref_ext:
no value
docref_root:
no value
enable_dl:
Off
enable_post_data_reading:
On
error_append_string:
no value
error_log:
no value
error_prepend_string:
no value
error_reporting:
22519
expose_php:
Off
extension_dir:
/opt/plesk/php/7.4/lib64/php/modules
file_uploads:
On
hard_timeout:
2
highlight.comment:
#FF8000
highlight.default:
#0000BB
highlight.html:
#000000
highlight.keyword:
#007700
highlight.string:
#DD0000
html_errors:
On
ignore_repeated_errors:
Off
ignore_repeated_source:
Off
ignore_user_abort:
Off
implicit_flush:
Off
include_path:
.:/opt/plesk/php/7.4/share/pear
input_encoding:
no value
internal_encoding:
no value
log_errors:
On
log_errors_max_len:
1024
mail.add_x_header:
Off
mail.force_extra_parameters:
no value
mail.log:
no value
max_execution_time:
120
max_file_uploads:
20
max_input_nesting_level:
64
max_input_time:
60
max_input_vars:
1000
memory_limit:
128M
open_basedir:
/var/www/vhosts/modx-3.amazili.com/:/tmp/
output_buffering:
4096
output_encoding:
no value
output_handler:
no value
post_max_size:
8M
precision:
14
realpath_cache_size:
4096K
realpath_cache_ttl:
120
register_argc_argv:
Off
report_memleaks:
On
report_zend_debug:
On
request_order:
GP
sendmail_from:
no value
sendmail_path:
/usr/sbin/sendmail -t -i
serialize_precision:
-1
short_open_tag:
Off
SMTP:
localhost
smtp_port:
25
sys_temp_dir:
no value
syslog.facility:
LOG_USER
syslog.filter:
no-ctrl
syslog.ident:
php
track_errors:
Off
unserialize_callback_func:
no value
upload_max_filesize:
2M
upload_tmp_dir:
no value
user_dir:
no value
user_ini.cache_ttl:
300
user_ini.filename:
.user.ini
variables_order:
GPCS
xmlrpc_error_number:
0
xmlrpc_errors:
Off
zend.assertions:
-1
zend.detect_unicode:
On
zend.enable_gc:
On
zend.exception_ignore_args:
On
zend.multibyte:
Off
zend.script_encoding:
no value
zend.signal_check:
Off

**Comment by Jako** [comment]
Please post the PHP error log message, that occurs during saving the resource.

**Comment by opengeek** [comment]
The error is this:

```[17-Jan-2022 09:42:21 America/Denver] PHP Fatal error:  Uncaught Error: Call to a member function get() on null in /Users/opengeek/www/revo/core/src/Revolution/Processors/Resource/Update.php:222
Stack trace:
#0 /Users/opengeek/www/revo/core/src/Revolution/Processors/Resource/Update.php(510): MODX\Revolution\Processors\Resource\Update->isSitePage('site_start')
#1 /Users/opengeek/www/revo/core/src/Revolution/Processors/Resource/Update.php(183): MODX\Revolution\Processors\Resource\Update->checkForUnPublishOnSitePages()
#2 /Users/opengeek/www/revo/core/src/Revolution/Processors/Model/UpdateProcessor.php(59): MODX\Revolution\Processors\Resource\Update->beforeSet()
#3 /Users/opengeek/www/revo/core/src/Revolution/Processors/Processor.php(189): MODX\Revolution\Processors\Model\UpdateProcessor->process()
#4 /Users/opengeek/www/revo/core/src/Revolution/modX.php(1771): MODX\Revolution\Processors\Processor->run()
#5 /Users/opengeek/www/revo/core/src/Revolution/modConnectorResponse.php(151): MODX\Revolution\modX->runProcessor('Resource/Update', Array, Array)
#6 /Users/opengeek in /Users/opengeek/www/revo/core/src/Revolution/Processors/Resource/Update.php on line 222
```

This code was merged in PR #15579, which was approved despite being completely non-functional. I have resolved it in commit a167399.

**Comment by AmaZili** [comment]
I can confirm this fix #15579 solved the issue
